### DISCUSSION TITLE
[Support] 属性装饰器的实现如果是个async函数，获取属性后是个promise，有什么途径能获取最终值吗

### DISCUSSION THREAD

**Discussion by minefujiko9293** [midwayjs/midway#3820] [Q&A]
### Describe the problem（描述问题）

![image](https://github.com/midwayjs/midway/assets/23348616/50d564f7-2ddc-402d-8167-9dc0cbb70fc0)

### Midway Versions（Midway 版本）

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)     
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.15.6)
✓ @midwayjs/bull(3.15.6)
✓ @midwayjs/bull-board(3.15.7)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)  
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.6)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed) 
✓ @midwayjs/cross-domain(3.15.7)
✓ @midwayjs/decorator(3.15.6)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(3.15.7)
✓ @midwayjs/jwt(3.15.7)
✓ @midwayjs/kafka(not installed) 
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed) 
✓ @midwayjs/mock(3.15.6)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)       
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.15.6)
✓ @midwayjs/security(not installed)   
✓ @midwayjs/sequelize(not installed)  
✓ @midwayjs/session(not installed)    
✓ @midwayjs/socketio(not installed)   
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)    
✓ @midwayjs/tablestore(not installed) 
✓ @midwayjs/tags(not installed)     
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.15.6)
✓ @midwayjs/upload(3.15.7)
✓ @midwayjs/validate(3.15.7)        
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.15.6)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment]
属性装饰器的实现方法不能是 promise，因为实际是做了 object.defineProperty 创建了 getter，这个时候是没有机会做异步操作的，所有的异步行为需要提前执行完成。
